accel/tcg/log_instr.c

```diff
@@ -1036,6 +1051,319 @@ void qemu_log_instr_extra(CPUArchState *env, const char *msg, ...)
     va_end(va);
 }
 
+// A printf that takes an array of argments unioned of all possible argument
+// types. Because we cant edit a VA_LIST, and we don't want the exponential blow
+// up of handling all combinations of types, we will bounce individual string
+// sections split in the fmt string to another buffer, then switch on all
+// possible types.
+static void g_string_append_printf_union_args(GString *string, const char *fmt,
+                                              qemu_log_arg_t *args)
+{
+
+// So Clang will not complain about the non-literal format.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-nonliteral"
+#pragma clang diagnostic ignored "-Wformat-security"
+    char bounce_buf[256];
+
+    size_t i = 0;
+    bool format = false;
+    char c;
+    bool is_short, is_long, is_long_long;
+    while ((c = bounce_buf[i++] = *fmt++)) {
+        if (!format) {
+            format = c == '%';
+            is_short = is_long = is_long_long = false;
+        } else {
+            bounce_buf[i] = '\0';
+            switch (c) {
+            case 'c':
+                g_string_append_printf(string, bounce_buf, (args++)->charv);
```

Comment:
Can we pull out all the args++ outside each statement? Also allows the if/else if/else if/else below to have a single args++.

Reply:
Kinda annoying, as some of the cases do not increment the pointer. I could always have those cases decrement the pointer, but that seems a little unnecessarily convoluted. I could hoist the args++ out if individual if/elses, but I'm not sure it would do anything for readability or performance.